app/shop.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+import datetime
+
+
+class Shop:
+    def __init__(
+            self,
+            name: str,
+            location: dict,
+            products: dict
```

Comment:
Product could also be presented as separate class and have method calcuclate_price